src/Workspaces/Core/Portable/Rename/Renamer.cs

```diff
@@ -25,6 +32,131 @@ public static class Renamer
                 newName, optionSet, cancellationToken);
         }
 
+        /// <summary>
+        /// Applies changes to the given document so that the folders, name, and path from <paramref name="newDocumentInfo"/>
+        /// are used. Will also perform symbol changes to the code in the document, attempting to sync type and namespace
+        /// to the new information.
+        /// 
+        /// Note that regardless of symbol changes, the document info will be applied and returned in the new solution. 
+        /// </summary>
+        /// <remarks>
+        /// Casing changes in document info will be applied but will not result in any symbol changes.
```

Comment:
Why not? Anybody who has had a debate of whether the type should be called "Intellisense" or "IntelliSense" might be surprised by this. :smile:

Reply:
(there appear to be many places in Roslyn where we have a file name with one casing and a folder name that contains the file with the other casing!)

Reply:
Right now it's because the inverse, casing changes to a type, do not apply changes to a filename. We can debate whether that should be true, but this has some precedence for behavior.